code/game/machinery/computer/atmos_computers/_air_sensor.dm

```diff
@@ -0,0 +1,64 @@
+/////////////////////////////////////////////////////////////
+// AIR SENSOR (found in gas tanks)
+/////////////////////////////////////////////////////////////
+
+/obj/machinery/air_sensor
+	name = "gas sensor"
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "gsensor1"
+	resistance_flags = FIRE_PROOF
+
+	var/on = TRUE
+
+	var/frequency = FREQ_ATMOS_STORAGE
+	var/datum/radio_frequency/radio_connection
+
+	/// The unique string that represents which atmos chamber to associate with.
+	var/chamber_id
+
+/obj/machinery/air_sensor/proc/broadcast_destruction(frequency)
+	var/datum/signal/signal = new(list(
+		"sigtype" = "destroyed",
+		"tag" = id_tag,
+		"timestamp" = world.time,
+	))
+	var/datum/radio_frequency/connection = SSradio.return_frequency(frequency)
+	connection.post_signal(null, signal, filter = RADIO_ATMOSIA)
+
+/obj/machinery/air_sensor/Destroy()
+	INVOKE_ASYNC(src, .proc/broadcast_destruction, src.frequency)
```

Comment:
it would be preferred to do this in code that reacts to the sensor being canonically "removed", rather then deleted.
This isn't a requirement mind, lord knows most of our code doesn't follow it, but it's often useful to keep side effects out of destroy. so destroy handles cleaning up any references to the object and nothing else, whereas something like `deconstruct` handles reacting to being say, beaten to death.

You'd still want the async, so the actual deletion isn't delayed by the signal sleeping

Reply:
Want me to split it into a new proc or want me to hook it into `deconstruct()`? Last time i checked `deconstruct()` don't always fire all the time, though i might be mistaken. Not really sure which proc to dump it into.